src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,108 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.command.*;
+import duke.exception.DukeException;
+import duke.exception.InvalidArgumentException;
+import duke.exception.InvalidCommandException;
+import duke.exception.InvalidTaskTypeException;
+import duke.task.*;
+
+/**
+ * The Parser class parses and processes user input.
+ */
+public class Parser {
+    /**
+     * Returns the appropriate {@link Command} to execute next.
```

Comment:
Should the javadoc comment header be more descriptive as the method is pretty long?